### PR TITLE
Added NR SP type to CorpMod

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#9244

*Description of changes:*
* Added NR specific Sole Prop type to CorpType mod
* To be used for local mapping of FR -> SP `nrTypeToEntityType()`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/modules/corp-type-module/corp-type-module.ts [hunk 1]
```diff
@@ -42,6 +42,7 @@ export enum CorpTypeCd {
   SOCIETY = 'S',
   SOCIETY_BRANCH = 'SB',
   SOLE_PROP = 'SP',
+  NR_SOLE_PROP = 'FR', // SPECIAL NAMEREQUEST-ONLY ENTITY TYPE
```
Comment (severinbeauvais) [inline src/modules/corp-type-module/corp-type-module.ts:45]: 1. Please keep this list sorted by code (see comment on line 7).
2. This new code needs an object in the CorpInfoArray below (for lookup purposes).

Comment (cameron-freshworks) [reply src/modules/corp-type-module/corp-type-module.ts]: Updated!

Comment (severinbeauvais) [reply src/modules/corp-type-module/corp-type-module.ts]: Object looks good.

I believe sort is by code not key :)

Comment (cameron-freshworks) [reply src/modules/corp-type-module/corp-type-module.ts]: 🤦  Sorry, will fix

Comment (cameron-freshworks) [reply src/modules/corp-type-module/corp-type-module.ts]: updated, only took me 12 commits and 3 cups of coffee to get them alphabetical. It must be Friday

**Review by severinbeauvais — APPROVED** [review]

**Pull request merged**